packages/protocol/lib/web3-utils.ts

```diff
@@ -387,6 +393,22 @@ export function getFunctionSelectorsForContract(contract: any, contractName: str
   return selectors
 }
 
+export function checkInheritance(baseContractName: string, derivativeContractArtifact: any, artifacts: any) {
```

Comment:
Looking at this function, it looks like it checks imports rather than inheritance. Could lead to errors if someone uses it in the future not knowing the implementation details. Either rename to `checkImports`, or preferably, use the `baseContracts` field of the `ContractDefinition` AST node instead of `ImportDirective`s during the DFS.

Reply:
I guess I'm not totally clear on the significance of the distinction here, but I'd prefer to avoid more substantive changes to this PR if possible since it's been sitting in review for a while and we only really scoped .1 points to get this merged this sprint. Renaming is super easy, but if rewriting the function feels worth doing we might need to adjust the scope and even push this into next sprint. Let me know what you think is best

Reply:
In general, imports will also include libraries and external contracts that are called or `CREATE`d by the contract, not only contracts that we inherit from. For a contract like `Initializable`(/`V2`), it's only purpose is to use it as a mixin, so it's probably fine to assume that import = inherit, but for maintainability think it's important to name the function correctly.

So I'd say for now fine to just leave as is with the new name, but would be nice to add a TODO/issue to change this to check inheritance (it wouldn't be much more complicated than what's happening currently with the BFS).

Reply:
Thanks 🙏 I added a TODO comment